packages/plugins/plugin-sheet/src/integrations/thread-ranges.ts

```diff
@@ -0,0 +1,101 @@
+import { debounce } from '@dxos/async';
+import { fullyQualifiedId } from '@dxos/react-client/echo';
+import { type DxGridElement, type DxGridPosition } from '@dxos/react-ui-grid';
+
+import { useSheetContext } from '../components';
+import { type CellAddress, type CompleteCellRange, inRange } from '../defs';
+import { SHEET_PLUGIN } from '../meta';
+
+export const completeCellRangeToThreadCursor = (range: CompleteCellRange): string => {
+  return `${range.from.col},${range.from.row},${range.to.col},${range.to.row}`;
+};
+
+export const parseThreadAnchorAsCellRange = (cursor: string): CompleteCellRange | null => {
+  const coords = cursor.split(',');
+  if (coords.length !== 4) {
+    return null;
+  } else {
+    const [fromCol, fromRow, toCol, toRow] = coords;
+    return {
+      from: { col: parseInt(fromCol), row: parseInt(fromRow) },
+      to: { col: parseInt(toCol), row: parseInt(toRow) },
+    };
+  }
+};
+
+export const useUpdateFocusedCellOnThreadSelection = (grid: MutableRefObject<DxGridElement | null>) => {
+  const { model, setActiveRefs } = useSheetContext();
+  const handleScrollIntoView: IntentResolver = useCallback(
+    ({ action, data }) => {
+      switch (action) {
+        case LayoutAction.SCROLL_INTO_VIEW: {
+          if (!data?.id || data?.cursor === undefined || data?.id !== fullyQualifiedId(model.sheet)) {
+            return;
+          }
+          setActiveRefs(data.thread);
+          // TODO(Zan): Everywhere we refer to the cursor in a thread context should change to `anchor`.
+          const range = parseThreadAnchorAsCellRange(data.cursor);
+          range && grid.current?.setFocus({ ...range.to, plane: 'grid' }, true);
+        }
+      }
+    },
+    [model.sheet, setActiveRefs],
+  );
+
+  useIntentResolver(SHEET_PLUGIN, handleScrollIntoView);
+};
+
+export const useSelectThreadOnCellFocus = () => {
+  const { model, cursor } = useSheetContext();
+  const dispatch = useIntentDispatcher();
+
+  const threads = useMemo(
+    () => model.sheet.threads?.filter((thread): thread is NonNullable<typeof thread> => !!thread) ?? [],
+    [
+      // TODO(thure): Surely we can find a better dependency for this…
+      JSON.stringify(model.sheet.threads),
+    ],
+  );
+
+  const selectClosestThread = useCallback(
+    (cellAddress: CellAddress) => {
+      if (!cellAddress || !threads) {
+        return;
+      }
+
+      const closestThread = threads?.find(({ anchor }) => {
+        if (anchor) {
+          const range = parseThreadAnchorAsCellRange(anchor);
+          return range ? inRange(range, cellAddress) : false;
+        } else {
+          return false;
+        }
+      });
+
+      if (closestThread) {
+        void dispatch([
+          { action: 'dxos.org/plugin/thread/action/select', data: { current: fullyQualifiedId(closestThread) } },
+        ]);
+      }
+    },
+    [dispatch, threads],
+  );
```

Comment:
We used to actively highlight in the sidebar whichever thread anchor was closest to the cursor, not just 1:1 overlap. This worked like the editor. Now we only select in the sidebar if we overlap?

Reply:
You could calculate distance to the range boundary fairly easily i'd say. It's a nice little sprinkle.

Reply:
Agreed this is potentially a helpful increment, worth filing as an additional issue?